apps/deploy-web/src/components/remote-deploy/remote-deploy-api-queries/bit-bucket-queries.ts

```diff
@@ -0,0 +1,208 @@
+import { useMutation, useQuery } from "react-query";
+import axios, { AxiosError } from "axios";
+import { useAtom } from "jotai";
+import { useRouter } from "next/navigation";
+
+import { tokens } from "@src/store/remoteDeployStore";
+import { BitBucketCommit } from "@src/types/remoteCommits";
+import { IGithubDirectoryItem, PackageJson } from "@src/types/remotedeploy";
+import { BitProfile } from "@src/types/remoteProfile";
+import { BitRepository, BitWorkspace } from "@src/types/remoteRepos";
+import { RouteStep } from "@src/types/route-steps.type";
+import { UrlService } from "@src/utils/urlUtils";
+import { ciCdTemplateId } from "../helper-functions";
+
+const Bitbucket_API_URL = "https://api.bitbucket.org/2.0";
+
+export const handleLoginBit = () => {
+  window.location.href = `https://bitbucket.org/site/oauth2/authorize?client_id=${process.env.NEXT_PUBLIC_BITBUCKET_CLIENT_ID}&response_type=code`;
+};
+const axiosInstance = axios.create({
+  baseURL: Bitbucket_API_URL,
+  headers: {
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  }
+});
+
+export const useFetchRefreshBitToken = () => {
+  const [token, setToken] = useAtom(tokens);
+
+  return useMutation({
+    mutationFn: async () => {
+      const response = await axios.post(`/api/bitbucket/refresh`, {
+        refreshToken: token?.refresh_token
+      });
+
+      return response.data;
+    },
+    onSuccess: data => {
+      setToken({
+        access_token: data.access_token,
+        refresh_token: data.refresh_token,
+        type: "bitbucket"
+      });
+    }
+  });
+};
+
+export const useBitFetchAccessToken = () => {
+  const [, setToken] = useAtom(tokens);
+
+  const router = useRouter();
+  return useMutation({
+    mutationFn: async (code: string) => {
+      const response = await axios.post(`/api/bitbucket/authenticate`, {
+        code
+      });
+
+      return response.data;
+    },
+    onSuccess: data => {
+      setToken({
+        access_token: data.access_token,
+        refresh_token: data.refresh_token,
+        type: "bitbucket"
+      });
+
+      router.replace(
+        UrlService.newDeployment({
+          step: RouteStep.editDeployment,
+          gitProvider: "github",
+          templateId: ciCdTemplateId
+        })
+      );
+    }
+  });
+};
+
+export const useBitUserProfile = () => {
+  const [token] = useAtom(tokens);
+  const { mutate } = useFetchRefreshBitToken();
+  return useQuery({
+    queryKey: ["userProfile", token.access_token],
```

Comment:
We have a `queryKeys` object where we set all the react-query keys, please move all keys over there, as well move all the queries in the queries folder.